static/js/services/translate.js

```diff
@@ -3,14 +3,19 @@
  */
 angular.module('inboxServices').service('Translate',
   function(
+    $log,
     $translate
   ) {
 
     'use strict';
     'ngInject';
 
     return function(translationId, interpolateParams) {
-      return $translate(translationId, interpolateParams);
+      return $translate(translationId, interpolateParams)
+        .catch(function() {
+          $log.warn('Translation service could not find a translation for ' + translationId);
```

Comment:
controversial opinion, but I really feel like `$translate` not finding a translation is not a crash-worthy event, even by default…

Reply:
Also, missing transitions are already logged by the $translation service so there's no need to log it out again.

Reply:
Sorry I guess my comment wasn't clear enough above: By default, if `$translate` can't find the key it throws an exception. You can solve this by passing a default value to the service, but to me this is stupid, because it should work, by default, in the same way it works when you template a page (ie your website rendering doesn't just break because you're missing a translation).

Happy to get rid of the log though

Reply:
I added a `Translate` service which wraps the `$translate` service mostly so we could mock it out for testing purposes. I agree that it shouldn't throw an Error on missing translation, so maybe modify the `Translate` service to catch the Error and use that instead?

Reply:
Ignore the above comment - I thought I was reading a different file. Looks good to me!